### DISCUSSION TITLE
How to make your own substitutions

### DISCUSSION THREAD

**Discussion by erap320** [erap320/LenovoWatchMOD#25] [Q&A] [answered]
Since I received requests of custom mods with particular notification apps, or questions on how to make substitutions, I'm making this issue to make relevant information on the subject easy to find.

You can read about it in #1, starting from [this comment](https://github.com/ERap320/LenovoWatchMOD/issues/1#issuecomment-406048546) in particular.

Please don't open issues to ask for custom mods.
I can't make them for everyone, so I won't make them for anyone.

Happy modding!

**Comment by xqwerty151x** [comment]
I looked at the post you referred to and I couldn't find where you could change what notifications you received. i have no experience doing any coding but from how your changes made it seem is that all you have to do is change the link in the code. I just want to add apps like Group Me and Discord to my notifications. What file is the code for what notifications come through stored?

**Comment by erap320** [comment, marked as answer]
@xqwerty151x quoting straight from #1:


> It mainly was swapping package names of chinese apps with the ones I wanted, as you can see in [Substitutions.txt](https://github.com/ERap320/LenovoWatchMOD/blob/master/Substitutions.txt). This happened in `smali_classes2\com\pmpd\business\device\notify\NotificationService.smali` and `smali_classes2\com\pmpd\business\device\notify\ConstantPackageName.smali`.

As said there, you have to take package names of the apps and change them with the ones you want.
The part of [Substitutions.txt](https://github.com/ERap320/LenovoWatchMOD/blob/master/Substitutions.txt) you need to look at is "Package names".

Obviously, if you're starting from my mod instead of the official app you have to find and substitute the apps I put in, not the chinese ones.

Those files are just the _"notification engine"_, so they don't affect any interface.
If you also want to  change the name of the apps in the UI accordingly to your substitutions, read more in #1

**Comment by ghost** [comment]
Hey , so I understand how I am supposed to change which I want to be changed , but what about the .txt ? How do I import it ? I have to say I don't know a lot about coding , especially on Android , but do I have to get the source code and recompiling it ?
Hope to have a answer soon :)

**Comment by erap320** [comment]
@Virtualexis The .txt just tells you the changes I made to my version of the MOD.
If you want to make your version you have to manually change the values in the .smali files I said before.
